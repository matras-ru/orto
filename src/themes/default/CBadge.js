export const base = 'rounded-lg font-bold px-0-6 py-0-2 inline-block border-2 leading-snug';

export const variantPrimary = 'bg-primary-100 border-primary-100 text-white';
export const variantSecondary = 'border-secondary-200 text-secondary-200';
export const variantTertiary = 'text-danger border-danger';
export const variantQuaternary = '';
export const variantQuinary = '';

export default {
    base,

    variantPrimary,
    variantSecondary,
    variantTertiary,
    variantQuaternary,
    variantQuinary
};
