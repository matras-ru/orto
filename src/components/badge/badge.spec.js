/* eslint-env jest */
import { mount } from '@vue/test-utils';
import CBadge from './badge';

const baseClass = 'rounded-lg font-bold inline-block border-2 align-middle leading-none';
const variantPrimaryClass = 'bg-primary-100 border-primary-100 text-white';
const variantSecondaryClass = 'border-secondary-200 text-secondary-200';

const sizeMd = 'text-base px-0-6 py-0-3';

describe('Badge', () => {
    it('The badge is rendered', () => {
        const wrapper = mount(CBadge);

        expect(wrapper.element.tagName).toEqual('SPAN');
        expect(wrapper.classes().sort()).toEqual(
            `${baseClass} ${sizeMd} ${variantPrimaryClass}`.split(' ').sort()
        );
        expect(wrapper.html()).toMatchSnapshot();

        wrapper.destroy();
    });

    it('The badge is rendered', () => {
        const wrapper = mount(CBadge, {
            context: {
                props: {
                    variant: 'secondary'
                }
            }
        });

        expect(wrapper.element.tagName).toEqual('SPAN');
        expect(wrapper.classes().sort()).toEqual(
            `${baseClass} ${sizeMd} ${variantSecondaryClass}`.split(' ').sort()
        );
        expect(wrapper.html()).toMatchSnapshot();

        wrapper.destroy();
    });
});
