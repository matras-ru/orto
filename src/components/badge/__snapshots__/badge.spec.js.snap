// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge The badge is rendered 1`] = `<span class="rounded-lg font-bold inline-block border-2 leading-none align-middle bg-primary-100 border-primary-100 text-white text-base px-0-6 py-0-3"></span>`;

exports[`Badge The badge is rendered 2`] = `<span class="rounded-lg font-bold inline-block border-2 leading-none align-middle border-secondary-200 text-secondary-200 text-base px-0-6 py-0-3"></span>`;
